test/app-luatest/fix_lua_loader_error_handling_test.lua

```diff
@@ -0,0 +1,24 @@
+local t = require('luatest')
```

Comment:
Oops, my bad. I've finally found out the reason why the test checks nothing. The fix is below. Furthermore, I've fixed all other your comments within this changeset.

```diff
diff --git a/test/app-luatest/fix_lua_loader_error_handling_test.lua b/test/app-luatest/fix_lua_loader_error_handling_test.lua
index 9aa552f6ea..f42f15abd3 100644
--- a/test/app-luatest/fix_lua_loader_error_handling_test.lua
+++ b/test/app-luatest/fix_lua_loader_error_handling_test.lua
@@ -1,24 +1,38 @@
 local t = require('luatest')
 local treegen = require('test.treegen')
+local fio = require('fio')
 
 local g = t.group()
 
+-- XXX: I would rather tweak <package.path>, since it's the
+-- recommended way to manage the hints for the loaders. However,
+-- the loader provided by LuaJIT that uses <package.path> works
+-- fine (i.e. throws the expected error), unlike the ones added
+-- for Tarantool. Hence, the easiest way to check Tarantool
+-- loaders is switching the current working directory for the
+-- test. Preserve the original working directory to restore it at
+-- the end of the test.
+local cwd = fio.cwd()
+
 local luadir
-local module = 'syntax'
+local name = 'syntax'
+local errlike = [[error loading module '<NAME>' from file '<DIR>/<NAME>.lua':
+       <DIR>/<NAME>.lua:1: unexpected symbol near '?']]
+
 
 g.before_all(function()
     treegen.init(g)
-    treegen.add_template(g, ('^%s%%.lua$'):format(module), [[?syntax error?]])
-    luadir = treegen.prepare_directory(g, {('%s.lua'):format(module)})
+    treegen.add_template(g, ('^%s%%.lua$'):format(name), [[?syntax error?]])
+    luadir = treegen.prepare_directory(g, {('%s.lua'):format(name)})
+    fio.chdir(luadir)
 end)
 
 g.after_all(function(g)
     treegen.clean(g)
+    fio.chdir(cwd)
 end)
 
 g.test_loader_error_handling = function()
-    package.path = ('%s/?.lua'):format(luadir)
-    local errlike = ("error loading module '%s' from file '%s/%s%%.lua':.+")
-                   :format(module, luadir, module)
-    t.assert_error_msg_matches(errlike, require, module)
+    local errmsg = errlike:gsub('%<(%w+)>', { NAME = name, DIR = luadir })
+    t.assert_error_msg_equals(errmsg, require, name)
 end
```